protocol/communities/manager.go

```diff
@@ -1681,7 +1681,33 @@ func (m *Manager) CheckPermissionToJoin(id []byte, addresses []gethcommon.Addres
 		}
 		return response, nil
 	}
-	return m.checkPermissions(permissionsToJoin, accountsAndChainIDs, false)
+	checkPermissionsResponse, err := m.checkPermissions(permissionsToJoin, accountsAndChainIDs, false)
```

Comment:
I think it can be done simpler


```
if len(becomeMemberPermissions) == 0 {
		// There is no permission to join, we can just say it's satisfied
		checkPermissionsResponse.Satisfied = true
		return checkPermissionsResponse, nil
}

return m.checkPermissions(permissionsToJoin, accountsAndChainIDs, true)
```

Reply:
Changing false to true is not the solution, it will only affect shortcircuit parameter `if shortcircuit is true, it will stop as soon as we know the answer`. If user is expecting full response, to check what type of permissions it satisfies exactly, then it will break its expectations.